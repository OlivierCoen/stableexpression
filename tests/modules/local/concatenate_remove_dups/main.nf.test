nextflow_process {

    name "Test Process CONCATENATE_REMOVE_DUPS"
    script "modules/local/concatenate_remove_dups/main.nf"
    process "CONCATENATE_REMOVE_DUPS"
    tag "concat"
    tag "module"

    test("Concat metadata files") {

        when {

            process {
                """
                input[0] = file('$projectDir/tests/input/concatenate_remove_dups/metadata1.csv')
                input[1] = file('$projectDir/tests/input/concatenate_remove_dups/metadata2.csv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with(process.out.merged) {
                assert path(get(0)).readLines().size() == 5
            }

        }

    }

    test("Concat mapping files") {

        when {

            process {
                """
                input[0] = file('$projectDir/tests/input/concatenate_remove_dups/mapping1.csv')
                input[1] = file('$projectDir/tests/input/concatenate_remove_dups/mapping2.csv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with(process.out.merged) {
                assert path(get(0)).readLines().size() == 5
            }

        }

    }

}
