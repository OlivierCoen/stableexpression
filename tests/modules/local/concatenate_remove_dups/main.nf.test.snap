{
    "Concat mapping files": {
        "content": [
            {
                "0": [
                    "merged.csv:md5,ecbc7c8bb8b78d7d3a20bbb08d86d56c"
                ],
                "1": [
                    [
                        "CONCATENATE_REMOVE_DUPS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "CONCATENATE_REMOVE_DUPS",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "merged": [
                    "merged.csv:md5,ecbc7c8bb8b78d7d3a20bbb08d86d56c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-02T13:00:31.910445267"
    },
    "Concat metadata files": {
        "content": [
            {
                "0": [
                    "merged.csv:md5,de2475f5a58b809a4e33af9789468eef"
                ],
                "1": [
                    [
                        "CONCATENATE_REMOVE_DUPS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "CONCATENATE_REMOVE_DUPS",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "merged": [
                    "merged.csv:md5,de2475f5a58b809a4e33af9789468eef"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-02T12:51:37.531039032"
    }
}